sycl/unittests/Extensions/CommandGraph/Barrier.cpp

```diff
@@ -875,19 +872,18 @@ TEST_F(CommandGraphTest, BarrierWithInOrderCommands) {
   // (5)    (6)
   //    \   /
   //    (B5)
-  auto GraphImpl = sycl::detail::getSyclObjImpl(Graph);
-  ASSERT_EQ(GraphImpl->MRoots.size(), 2lu);
+  experimental::detail::graph_impl &GraphImpl = *getSyclObjImpl(Graph);
+  ASSERT_EQ(GraphImpl.MRoots.size(), 2lu);
 
-  for (auto Root : GraphImpl->MRoots) {
+  for (auto Root : GraphImpl.MRoots) {
     auto RootNode = Root.lock();
     bool EvenPath;
 
     ASSERT_EQ(RootNode->MSuccessors.size(), 1lu);
-    if (GraphImpl->getEventForNode(RootNode) ==
-        sycl::detail::getSyclObjImpl(Node2)) {
+    if (GraphImpl.getEventForNode(RootNode).get() == &*getSyclObjImpl(Node2)) {
       EvenPath = true;
-    } else if (GraphImpl->getEventForNode(RootNode),
-               sycl::detail::getSyclObjImpl(Node1)) {
```

Comment:
This `,` doesn't make much sense to me, I assume it was supposed to be `==` (implemented on the right now). @intel/sycl-graphs-reviewers , @Bensuo can you confirm?

Reply:
Local build passed with my change, so I hope CI will work this time. Thanks for a quick reply!